bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -223,6 +225,61 @@ def jit_function(self, func, cols_list, extra_args):
         FunctionJitter.function_cache[self.func.__name__] = (self.func_call, self.func_sign)
         return self.func_call
 
+def _convert_to_vector(args):
+    """
+    Converts a Python list of strings into an std::vector before passing such
+    argument to a Filter operation.
+    The purpose of this function is to workaround issue #10092 until there is
+    a proper fix in cppyy.
+    Arguments:
+        args: arguments passed to Filter, possibly including the list of column
+    names
+    Returns:
+        Tuple with arguments, possibly replacing a list of column names by a
+    vector
+    """
+
+    import ROOT
+
+    if not isinstance(args[0], list): return args
+
+    try:
+        v = ROOT.std.vector['std::string'](args[0])
+    except TypeError:
+        raise TypeError(
+            f"The list of columns of a Filter operation can only contain strings. Please check: {args[0]}")
+
+    return v, *args[1:]
+
+def _handle_cpp_callables(func, original_template, *args):
+    """
+    Checks whether the callable `func` is a cppyy proxy of one of these:
+    1. C++ functor
+    2. std::function
+
+    The cases above are supported by cppyy, so we can just invoke the original
+    cppyy TemplateProxy (Filter or Define) with the callable as argument.
+
+    Prior to the invocation of the original cppyy TemplateProxy, though, we
+    need to explicitly instantiate the template using the type of the `func`
+    callable. Implicit instantiation won't work since the original
+    implementation of Filter and Define is replaced by a pythonization.
+
+    Arguments:
+        func: callable to be checked
+        original_rdf_method: cppyy proxy for Filter or Define
+        args: arguments passed by the user to Filter or Define
+    Returns:
+        RDataFrame: RDataFrame node if `func` can be classified in one of the
+    three cases above, None otherwise
+    """
+
+    is_cpp_functor  = lambda : type(getattr(func, '__call__', None)) == libcppyy.CPPOverload
+    is_std_function = lambda : isinstance(getattr(func, 'target_type', None), libcppyy.CPPOverload)
+
+    if is_cpp_functor() or is_std_function():
```

Comment:
Maybe a more concise way:
```suggestion
    target = getattr(func, 'target_type', None) or getattr(func, '__call__', None)
    if isinstance(target, libcppyy.CPPOverload):
```

Reply:
I acknowledge my code is a bit more verbose, but the purpose here was for future readers to understand what those conditions mean. And they are defined as lambdas so they are lazily evaluated.

Reply:
Yes sounds good, I would still change 
```suggestion
    is_cpp_functor  = lambda : isinstance(getattr(func, '__call__', None), libcppyy.CPPOverload)
    is_std_function = lambda : isinstance(getattr(func, 'target_type', None), libcppyy.CPPOverload)
```
For more consistency

Reply:
Will do, thanks!